src/runtime_view_proxy.js

```diff
@@ -0,0 +1,187 @@
+      const len = end - start
+      const targetArray = getHeapBlock(type, target * bpe, len)
+      const sourceArray = getHeapBlock(type, start * bpe, len)
+      targetArray.set(sourceArray)
+      return heapBlocks[0]
+    } else {
+      return heapBlocks[0].copyWithin(target, start, end)
+    }
+  }
+
+  function setOverridden(array, offset) {
+    const offsetReal = (offset || 0) * bpe
+    if (offsetReal >= maxArraySize || array.byteLength + offsetReal >= maxArraySize) {
+      const targetArray = getHeapBlock(type, offsetReal, array.length)
+      targetArray.set(array)
+    } else {
+      firstHeapBlock.set(array, offset)
+    }
+  }
+
+  function subarray(start, end) {
+    const [startReal, endReal] = getRealStartAndEnd(start, end)
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      return getHeapBlock(type, startReal, endReal - startReal)
+    } else {
+      return firstHeapBlock.subarray(start, end)
+    }
+  }
+
+  function fill(value, start, end) {
+    const [startReal, endReal] = getRealStartAndEnd(start, end)
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      const hb = getHeapBlock(type, startReal, endReal - startReal)
+      hb.fill(value, 0, end - start)
+      return firstHeapBlock
+    } else {
+      return firstHeapBlock.fill(value, start, end)
+    }
+  }
+  function slice(start, end) {
+    const [startReal, endReal] = getRealStartAndEnd(start, end)
+    if (startReal >= maxArraySize || endReal >= maxArraySize) {
+      const hb = getHeapBlock(type, startReal, endReal - startReal)
+      return hb.slice(start, end)
+    } else {
+      return firstHeapBlock.slice(start, end)
+    }
+  }
+
+  return {
+    get(target, property) {
+      if (parseInt(property, 10) == property) {
+        const memoryOffset = property * bpe
+        const blockNumber = Math.floor(memoryOffset / maxArraySize)
+        return heapBlocks[blockNumber][property - blockNumber * maxArraySize]
+      }
+
+      if (property === 'copyWithin') {
+        return copyWithin
+      }
+
+      if (property === 'set') {
+        return setOverridden
+      }
+
+      if (property === 'subarray') {
+        return subarray
+      }
+
+      if (property === 'fill') {
+        return fill
+      }
+
+      if (property === 'slice') {
+        return slice
+      }
+
+      return firstHeapBlock[property]
+    },
+    set(target, property, value) {
+      if (parseInt(property, 10) == property) {
+        const memoryOffset = property * bpe
+        const blockNumber = Math.floor(memoryOffset / maxArraySize)
+        heapBlocks[blockNumber][property - blockNumber * maxArraySize] = value
+        return true
+      }
+
+      firstHeapBlock[property] = value
+      return true;
+    },
+  }
+}
+
+function createMemoryProxy(type) {
+  const heapBlocks = [
+    getHeapBlock(type, 0),
+  ];
+  const numberOfBlocks = Math.ceil(b.byteLength / maxArraySize)
+  for (let i = 1; i < numberOfBlocks; i++) {
```

Comment:
Why not initialize `heapBlocks` to the empty array and than start this loop at 0?